pageserver/src/http/routes.rs

```diff
@@ -1463,6 +1463,45 @@ async fn timeline_layer_scan_disposable_keys(
     )
 }
 
+async fn timeline_download_heatmap_layers_handler(
+    request: Request<Body>,
+    _cancel: CancellationToken,
+) -> Result<Response<Body>, ApiError> {
+    let tenant_shard_id: TenantShardId = parse_request_param(&request, "tenant_shard_id")?;
+    let timeline_id: TimelineId = parse_request_param(&request, "timeline_id")?;
+    let concurrency: usize = parse_query_param(&request, "concurrency")?.unwrap_or(16);
```

Comment:
Maybe we should clamp `concurrency` to something sane like the remote storage concurrency semaphore?  Otherwise a rogue caller could pass a huge number that really dominates remote storage until all its downloads are complete

Reply:
https://github.com/neondatabase/neon/pull/10788/commits/f693efa458bc01eea933f3dab00a6c6e81060356